lib/src/transformer/declaration_parsing.dart

```diff
@@ -53,11 +53,20 @@ class ParsedDeclarations {
       key_stateMixin:        <CompilationUnitMember>[],
     };
 
+    var privatePropsAndStateClassPattern = new RegExp(r'^.*class\s(_[$].*?\s)');
```

Comment:
Instead of allowing anything before the `class` keyword, we should be more specific to avoid a commented-out class declaration from matching. Also changed up the class name capturing a bit:

```suggestion
    var privatePropsAndStateClassPattern = new RegExp(r'^(?:abstract )?class\s(_\$\w*?)\s');
```